### PR TITLE
Update buildbot_steps lint to handle env variables

### PR DESCRIPTION
**Pull request by aneeshusa**
https://github.com/servo/saltfs/pull/687 added support
for specifying environment variables in `buildbot_steps.yml`.
Update the servo-tidy buildbot_steps.yml linter to reflect this.

Use the voluptuous Python library (BSD 3-clause license) for validation
in lieu of a much larger hand-written implementation.

Extracted out of #17171. Helps with servo/saltfs#770.

- [ ] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because they change the tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19720)

### CODE REVIEW

**Comment by aneeshusa** [comment]
cc @jdm

**Comment by aneeshusa** [comment]
I've added a second commit which redoes #19612 (denying warnings in CI) with this new support as an example of its usefulness.

**Comment by aneeshusa** [comment]
Fixed tidy's self test and updated `.travis.yml` and `appveyor.yml` as well.

**Comment by jdm** [comment]
This is a nice improvement. Thanks!
@bors-servo r+

File: etc/ci/buildbot_steps.yml [hunk 1]
```diff
@@ -32,11 +35,11 @@ mac-rel-wpt4:
...
   - ./mach clean-nightlies --keep 3 --force
-  - env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig RUSTFLAGS=-Dwarnings ./mach build --dev
-  - env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig RUSTFLAGS=-Dwarnings ./mach test-unit
+  - env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig ./mach build --dev
+  - env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig ./mach test-unit
   - env ./mach package --dev
-  - env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig RUSTFLAGS=-Dwarnings ./mach build-cef
-  - env RUSTFLAGS=-Dwarnings ./mach build-geckolib
+  - env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig ./mach build-cef
+  - env ./mach build-geckolib
```
Comment (jdm) [inline etc/ci/buildbot_steps.yml]: We'll need to remove all of the now-unnecessary env prefixes.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by SimonSapin** [comment]
@aneeshusa I’ve pushed a fixup commit to your branch

@bors-servo try

**Comment by SimonSapin** [comment]
```
env CC=gcc-5 CXX=g++-5 "" ./mach build-cef
[…]
env: : No such file or directory
```

That `""` comes form having two consecutive spaces in `etc/ci/buildbot_steps.yml`, with the empty string between them being interpreted as a separate argument. Some bit of Python in servo/saltfs probably need to use `foo.split()` instead of `foo.split(" ")`, or something like that. I’ve removed the extra space.

**Pull request merged**